pkg/controller/garbagecollector/graph_builder.go

```diff
@@ -161,6 +163,17 @@ func (gb *GraphBuilder) informerFor(resource schema.GroupVersionResource, kind s
 			}
 			gb.graphChanges.Add(event)
 		},
+		ErrorFunc: func(err error) {
+			klog.Warningf("GB informerFor, got error for resource %q", resource)
```

Comment:
I could do that, but I just added this to demonstrate how `OnError` could be used. 

I was under the impression that we weren't going to use the errors at all in GC/RQ (or anywhere in k/k for that matter), because I remember you mentioning that we should _only_ be removing event handlers based on their disappearance from discovery (even though it would be quicker to remove them here when the error gets pushed to the event handlers).

Correct me if I'm wrong, but this `OnError` mechanism is primarily for use by controller-runtime (or anyone else that wants to use the watch error to trigger event handler removal instead of waiting for something a poll of discovery.

Reply:
> I just added this to demonstrate how `OnError` could be used.

oh, got it